### PR TITLE
Update Docker images

### PR DESCRIPTION
**Pull request by evanxg852000**
### Description
Closes https://github.com/quickwit-inc/quickwit/issues/972
- update docker image for new installation layout, 
- added/overridden image metadata 

### How was this PR tested?
Used custom build & install script to test.

### CODE REVIEW

**Comment by fmassot** [comment]
>  Though we don't build nightly, this image is currently used this way

I don't really get what you mean? The script is called each time there is a new commit on the main branch?

**Comment by evanxg852000** [comment]
I mean: The image is built every time we merge to `main`, this is not exactly nightly (cron scheduled build) but it's ok.

**Review by fmassot — APPROVED** [review]
LGTM.

a nit that you can ignore: I would prefer keeping main in the script name but I may not fully understand why you wanted to call it nightly

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
-LABEL maintainer="Quickwit, Inc. <hello@quickwit.io>"
+FROM alpine:3.14 AS builder
 
 # Install tools
 RUN apk add --no-cache curl
-
-# Install Quickwit
 RUN curl -L https://install.quickwit.io | sh
-RUN mv ./quickwit /usr/local/bin/quickwit
+RUN mv  ./quickwit-*/* ./
```
Comment (guilload) [inline docker/Dockerfile:6]: Wonder if we should `sed` the config here to set the listen address to `0.0.0.0`.
@fmassot ?

**Review by guilload — APPROVED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (evanxg852000) [reply docker/Dockerfile:6]: From the feedback on the other PR, we have a docker flavor of the config. We can copy that file too?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,14 +1,22 @@
-FROM alpine:3.14
-LABEL maintainer="Quickwit, Inc. <hello@quickwit.io>"
+FROM alpine:3.14 AS builder
```
[the remark this answers is not in the document]
Comment (evanxg852000) [reply docker/Dockerfile:1]: Well spotted :+1:

File: docker/Dockerfile (hunk 1 above)
Comment (fmassot) [reply docker/Dockerfile:6]: @guilload I suggested having a default config file suited for docker. 
So in this case we have the right config file from the start. Sed is fine too, I'm ok with both solutions.

**Pull request merged**